wormchain/interchaintest/helpers/cw_wormhole/helpers.go

```diff
@@ -0,0 +1,408 @@
+	govVaa := helpers.GenerateGovernanceVaa(uint32(guardianSetIndex), guardians, payload)
+	vaaBz, err := govVaa.Marshal()
+	require.NoError(t, err)
+
+	encodedVaa := base64.StdEncoding.EncodeToString(vaaBz)
+	executeVAAPayload, err := json.Marshal(ExecuteMsg{
+		SubmitVaa: &ExecuteMsg_SubmitVAA{
+			Vaa: Binary(encodedVaa),
+		},
+	})
+	require.NoError(t, err)
+
+	// Submit VAA
+	_, err = chain.ExecuteContract(ctx, keyName, contractAddr, string(executeVAAPayload))
+	return err
+}
+
+// SubmitFeeUpdate submits a VAA to update the fee amount
+func SubmitFeeUpdate(
+	t *testing.T,
+	ctx context.Context,
+	guardians *guardians.ValSet,
+	wormchain *cosmos.CosmosChain,
+	contractAddr string,
+	amount string,
+	replay bool,
+) (*TxResponse, error) {
+	// Get current guardian set index
+	guardianSetIndex := helpers.QueryConsensusGuardianSetIndex(t, wormchain, ctx)
+
+	// Create a fixed 32-byte array for the fee amount
+	var amountBytes [32]byte
+	amountInt := new(big.Int)
+	amountInt.SetString(amount, 10)
+	amountInt.FillBytes(amountBytes[:])
+
+	// Create governance VAA payload
+	// [32 bytes] Core Module
+	// [1 byte]  Action (3 for set fee)
+	// [2 bytes] ChainID (0 for universal)
+	// [32 bytes] Amount
+	buf := new(bytes.Buffer)
+	buf.Write(vaa.CoreModule)
+	vaa.MustWrite(buf, binary.BigEndian, vaa.ActionCoreSetMessageFee)
+	vaa.MustWrite(buf, binary.BigEndian, uint16(0)) // ChainID 0 for universal
+	buf.Write(amountBytes[:])
+
+	// Generate and sign governance VAA
+	govVaa := helpers.GenerateGovernanceVaa(uint32(guardianSetIndex), guardians, buf.Bytes())
+	vaaBz, err := govVaa.Marshal()
+	require.NoError(t, err)
+
+	// Rest of the function remains the same...
+	encodedVaa := base64.StdEncoding.EncodeToString(vaaBz)
+	executeVAAPayload, err := json.Marshal(ExecuteMsg{
+		SubmitVaa: &ExecuteMsg_SubmitVAA{
+			Vaa: Binary(encodedVaa),
+		},
+	})
+	require.NoError(t, err)
+
+	// Submit VAA
+	txHash, err := wormchain.ExecuteContract(ctx, "faucet", contractAddr, string(executeVAAPayload))
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait for transaction
+	err = testutil.WaitForBlocks(ctx, 2, wormchain)
+	require.NoError(t, err)
+
+	// Replay the transaction
+	if replay {
+		_, err = wormchain.ExecuteContract(ctx, "faucet", contractAddr, string(executeVAAPayload))
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "VaaAlreadyExecuted")
+	}
+
+	// Query transaction result
+	txResult, _, err := wormchain.Validators[0].ExecQuery(ctx, "tx", txHash)
+	require.NoError(t, err)
+
+	// Parse response
+	var txResponse TxResponse
+	err = json.Unmarshal(txResult, &txResponse)
+	require.NoError(t, err)
+
+	return &txResponse, nil
+}
+
+// VerifyFee verifies the fee amount in the contract state
+func VerifyFee(
+	t *testing.T,
+	ctx context.Context,
+	wormchain *cosmos.CosmosChain,
+	contractAddr string,
+	expectedAmount string,
+) {
+	var stateResp GetStateQueryResponse
+	err := wormchain.QueryContract(ctx, contractAddr, QueryMsg{
```

Comment:
i think querying the fee is good, but it would be better if we also attempt to submit a tx with a lower fee and ensure that it fails and using the new fee succeeds.